### ISSUE TITLE
🐛 `lint/correctness/noUnreachableSuper` triggers a false positive if the constructor contains a loop

### ISSUE THREAD

**Issue by lgarron** [rome/tools#4615]
### Environment information

```block
(Excluded, as `npx rome rage` outputs a suuuuper verbose error related to LSP, and this is reproducible in the playground. This is reproducible using the CLI version `12.1.3-nightly.50eb45f`.)
```

### What happened?

Use the following code:

```js
class A {
  constructor() {
    console.log("Constructing A!")
  }
}

class B extends A {
  constructor() {
    super();

    for (const i of [1, 2]) {
      console.log("Counting:", i)
    }
  }
}
```
https://docs.rome.tools/playground/?code=YwBsAGEAcwBzACAAQQAgAHsACgAgACAAYwBvAG4AcwB0AHIAdQBjAHQAbwByACgAKQAgAHsACgAgACAAIAAgAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoACIAQwBvAG4AcwB0AHIAdQBjAHQAaQBuAGcAIABBACEAIgApAAoAIAAgAH0ACgB9AAoACgBjAGwAYQBzAHMAIABCACAAZQB4AHQAZQBuAGQAcwAgAEEAIAB7AAoAIAAgAGMAbwBuAHMAdAByAHUAYwB0AG8AcgAoACkAIAB7AAoAIAAgACAAIABzAHUAcABlAHIAKAApADsACgAgACAAIAAgAAoAIAAgACAAIABmAG8AcgAgACgAYwBvAG4AcwB0ACAAaQAgAG8AZgAgAFsAMQAsACAAMgBdACkAIAB7AAoAIAAgACAAIAAgACAAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAIgBDAG8AdQBuAHQAaQBuAGcAOgAiACwAIABpACkACgAgACAAIAAgAH0ACgAgACAAfQAKAH0A

This errors with `lint/correctness/noUnreachableSuper` using the message: <code>This constructor calls `super()` in a loop.</code>

### Expected result

No error, since the loop happens after the (sole) `super()` call.

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

**Comment by nissy-dev** [comment]
I think this issue was fixed by https://github.com/rome/tools/pull/4614

**Comment by lgarron** [comment]
> I think this issue was fixed by #4614

Thanks! That PR seems to be specifically about `if` statements — do you think it will also cover the `for` loop case?

**Comment by nissy-dev** [comment]
Yes, you can confirm that the issue was fixed in the playground.
